source/features/restore-file.tsx

```diff
@@ -86,43 +88,72 @@ async function discardChanges(progress: (message: string) => void, originalFileN
 
 async function handleClick(event: DelegateEvent<MouseEvent, HTMLButtonElement>): Promise<void> {
 	const menuItem = event.delegateTarget;
+	const filenames = getFilenames(menuItem);
 
-	const [originalFileName, newFileName = originalFileName] = menuItem
-		.closest('[data-path]')!
-		.querySelector('.Link--primary')!
-		.textContent
-		.split(' → ');
+	const commitTitle = prompt('Are you sure you want to discard these changes? Enter the commit title', `Discard changes to ${filenames.original}`);
 
-	const commitTitle = prompt(`Are you sure you want to discard the changes to ${newFileName}? Enter the commit title`, `Discard changes to ${newFileName}`);
 	if (!commitTitle) {
 		return;
 	}
 
-	await showToast(async progress => discardChanges(progress!, originalFileName, newFileName, commitTitle), {
+	await showToast(async progress => discardChanges(progress!, filenames.original, filenames.new, commitTitle), {
 		message: 'Loading info…',
 		doneMessage: 'Changes discarded',
 	});
 
 	// Hide file from view
-	menuItem.closest('.file')!.remove();
+	const filesWrapper = $('div[class^="prc-PageLayout-Content-"] div[data-hpc="true"]');
+	if (filesWrapper) {
+		const fileElement = [...filesWrapper.children].find(child =>
+			child.textContent?.includes(filenames.original),
+		) as HTMLElement | undefined;
+
+		if (fileElement) {
+			fileElement.remove();
+		}
+	} else {
+		menuItem.closest('.file')!.remove();
+	}
 }
 
 function add(editFile: HTMLAnchorElement): void {
-	editFile.after(
-		<button
-			className="pl-5 dropdown-item btn-link rgh-restore-file"
-			role="menuitem"
-			type="button"
-		>
-			Discard changes
-		</button>,
-	);
+	const content = editFile.tagName === 'LI'
```

Comment:
Instead of assigning it, just attach it. Also place the old one in a condition so it can be removed cleanly next year 

```js
// TODO: Drop support for old view in June 2025
if (editFile.tagName === 'A') {
	editFile.after(
		<button
			className="pl-5 dropdown-item btn-link rgh-restore-file"
			role="menuitem"
			type="button"
		>
			Discard changes
		</button>
	);

  return;
}

etc
```

Reply:
> Also place the old one in a condition so it can be removed cleanly next year

This was not addressed and it applies to both conditions. It should ideally look like this:


```js
if (condition) {
  // old code
  return
}

// new code
```

so next year we apply this diff:

```diff
- if (condition) {
-   // old code
-   return
- }

  // new code
```

and you keep the git-blame for that `// new code` line.

Also keep this in mind: https://eslint.org/docs/latest/rules/no-else-return (our lint config is broken right now)